### PR TITLE
tailscale-systemd package

### PR DESCRIPTION
**Pull request by kiwiz**
*What*
A package with the necessary systemd service configuration to get Tailscale working on rM devices. Requires the base `tailscale` package.

*Authorship*
Service file comes directly from https://remarkable.guide/tailscale.html

*Specificity*
Should work on all

### CODE REVIEW

File: package/tailscale-systemd/package [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+url=https://tailscale.com
+pkgver=0.0.1
+section="devel"
+timestamp=2023-07-12T00:00Z
+maintainer="Kai <z@kwi.li>"
+license=MIT
+installdepends=(tailscale)
+
+package() {
+    install -D -m 644 -t "$srcdir"/tailscaled.service /lib/systemd/system/tailscaled.service
```
Comment (Eeems) [inline package/tailscale-systemd/package]: We've been putting service files under `/etc/systemd/system` as a rule in toltec.

File: package/tailscale-systemd/tailscaled.service [hunk 2]
```diff
@@ -0,0 +1,24 @@
+[Unit]
+Description=Tailscale node agent
+Documentation=https://tailscale.com/kb/
+Wants=network-pre.target
+After=network-pre.target
+
+[Service]
+Environment="HOME=/home/root"
+ExecStartPre=/opt/bin/tailscaled --cleanup
+ExecStart=/opt/bin/tailscaled --state=/var/lib/tailscale/tailscaled.state --socket=/run/tailscale/tailscaled.sock --tun=userspace-networking --socks5-server=localhost:1055 --outbound-http-proxy-listen=localhost:1055
```
Comment (Eeems) [inline package/tailscale-systemd/tailscaled.service]: Try to place things under `/opt` if possible. So `/var/lib/tailscale/tailscaled.state` should be `/opt/var/lib/tailscale/tailscaled.state`. it's fine to use `/run` as it's tmpfs.

**Review by Eeems — CHANGES_REQUESTED** [review]

**Comment by kiwiz** [comment]
* [ ] Test on device

**Comment by kiwiz** [comment]
Having trouble running `toltecmk` on my host. Looks like it might be due to https://github.com/toltec-dev/build/pull/31, but I don't see anything that would be interpretted as a C-quoted string.

**Comment by Eeems** [comment]
> Having trouble running `toltecmk` on my host. Looks like it might be due to [toltec-dev/build#31](https://github.com/toltec-dev/build/pull/31), but I don't see anything that would be interpretted as a C-quoted string.

It's based on the version of bash you are using underneath or something IIRC, so you won't see it in your package recipe.

File: package/tailscale-systemd/package [hunk 3]
```diff
@@ -1,6 +1,6 @@
 #!/usr/bin/env bash
 # Copyright (c) 2020 The Toltec Contributors
-# SPDX-License-Identifier: MIT
+# SPDX-License-Identifier: BSD-3-Clause
```
Comment (Eeems) [inline package/tailscale-systemd/package]: This is the license for this file, not the underlying software it's packaging. It must be MIT.

**Comment by kiwiz** [comment]
Tested on my rm2. Install + usage works fine. However, `is-active` and `is-enabled` are undefined when uninstalling. Do those need to be sourced somehow?

**Comment by Eeems** [comment]
> Tested on my rm2. Install + usage works fine. However, `is-active` and `is-enabled` are undefined when uninstalling. Do those need to be sourced somehow?

When built through toltec, install-lib is available, but toltecmk doesn't include it.

I work around that when building with toltecmk by embedding them when a package isn't part of the main toltec workflow: https://github.com/Eeems-Org/oxide/blob/master/package#L17-L53

**Comment by kiwiz** [comment]
Ack. Tested locally with the relevant functions and it looks good.

**Comment by Eeems** [comment]
> Ack. Tested locally with the relevant functions and it looks good.

Have you tested with the packages generated by the actions on this PR?

**Comment by kiwiz** [comment]
Oh, I have not! Will do that now.

**Comment by kiwiz** [comment]
Confirmed that:
1) Package installs
2) Tailscaled comes up
3) Uninstall cleans up service

**Review by Eeems — APPROVED** [review]
I haven't tested anything as I'm not setup to be able to test tailscale at all, but if it works for you that's good enough for me right now.

**Pull request merged**